### DISCUSSION TITLE
Build error on current git version - module not supported

### DISCUSSION THREAD

**Discussion by zyklopisch** [vercel/next.js#46119] [Help]
When I am running npm run build, I get this error:

Build error occurred
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/cadayton/thunderhub/node_modules/d3-scale/src/index.js from /home/cadayton/thunderhub/node_modules/@visx/scale/lib/scales/band.js not supported.
Instead change the require of index.js in /home/cadayton/thunderhub/node_modules/@visx/scale/lib/scales/band.js to a dynamic import() which is available in all CommonJS modules.

I have done some research and it seems that next.js (the version installed) may cause the problem - can you help me to fix this? Or do you have any idea on how to downgrade this and get it to work?

**Comment by icyJoseph** [comment]
I think you need to use transpile modules on that d3-scale library.  https://nextjs.org/docs/advanced-features/compiler#module-transpilation

What Next.js version do you have?

**Comment by zyklopisch** [comment]
Thx - in the meanwhile, I tried to downgrade to an old version of nodejs and npm - after that it worked magically.... seems it was somehow package related before.
